orderer/common/cluster/util.go

```diff
@@ -823,3 +827,38 @@ func EncodeTimestamp(t *timestamppb.Timestamp) []byte {
 	binary.LittleEndian.PutUint64(b, uint64(t.Seconds))
 	return b
 }
+
+// ExtractPublicKeyFromCert extracts the public key from an X.509 certificate
+func ExtractPublicKeyFromCert(der []byte) ([]byte, error) {
+	cert, err := x509.ParseCertificate(der)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to parse certificate")
+	}
+
+	return x509.MarshalPKIXPublicKey(cert.PublicKey)
+}
+
+func CompareCertPublicKeys(cert1, cert2 []byte) (bool, error) {
+	// Extract public key using the same approach as IsConsenterOfChannel
```

Comment:
Note that IsConsenterOfChannel with BFT compares enrollment certs, it is the raft version of IsConsenterOfChannel that compares public keys.

Reply:
That's what he's doing here, he's changing the implementation to fit the Raft style where we only care about the public key, not the entire cert. So that if you need to rotate your enrollment certificate, you won't need to do a channel config.